modules/widgets/src/widgets/reset-view-widget.tsx

```diff
@@ -0,0 +1,81 @@
+// deck.gl
+// SPDX-License-Identifier: MIT
+// Copyright (c) vis.gl contributors
+
+/* global document */
+import {WidgetImpl, WidgetImplProps} from '../widget-impl';
+import {h, render} from 'preact';
+import {IconButton} from '../components';
+import type {WidgetPlacement} from '@deck.gl/core';
+
+/** @todo - is the the best we can do? */
+type ViewState = Record<string, unknown>;
+
+/** Properties for the ResetViewWidget */
+export type ResetViewWidgetProps = WidgetImplProps & {
+  /** Widget positioning within the view. Default 'top-left'. */
+  placement?: WidgetPlacement;
+  /** Tooltip message */
+  label?: string;
+  /** The initial view state to reset the view to. Defaults to deck.props.initialViewState */
+  initialViewState?: ViewState;
+  /** View to interact with. Required when using multiple views. */
+  viewId?: string | null;
+};
+
+/**
+ * A button widget that resets the view state of deck to an initial state.
+ */
+export class ResetViewWidget extends WidgetImpl<ResetViewWidgetProps> {
+  static defaultProps: Required<ResetViewWidgetProps> = {
+    ...WidgetImpl.defaultProps,
+    id: 'reset-view',
+    placement: 'top-left',
+    label: 'Reset View',
+    initialViewState: undefined!,
+    viewId: undefined!
+  };
+
+  className = 'deck-widget-reset-view';
+  placement: WidgetPlacement = 'top-left';
+
+  constructor(props: ResetViewWidgetProps = {}) {
+    super({...ResetViewWidget.defaultProps, ...props});
+    this.placement = props.placement ?? this.placement;
+  }
+
+  setProps(props: Partial<ResetViewWidgetProps>) {
+    this.placement = props.placement ?? this.placement;
+    super.setProps(props);
+  }
+
+  onRenderHTML() {
+    const element = this.element;
+    if (!element) return;
+    render(
+      <IconButton
+        className="deck-widget-reset-focus"
+        label={this.props.label}
+        onClick={this.handleClick.bind(this)}
+      />,
+      element
+    );
+  }
+
+  handleClick() {
+    const initialViewState = this.props.initialViewState || this.deck?.props.initialViewState;
+    this.setViewState(initialViewState);
+  }
+
+  setViewState(viewState: ViewState) {
```

Comment:
I think we should make methods for controlling deck available, perhaps on the widgetManager, so that we can hide the ugly hacks we are doing now and evolve a proper API to make it easy for widget writers.